File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/CDDriveChassis.java
package org.firstinspires.ftc.teamcode;


import com.qualcomm.robotcore.hardware.DcMotor;
import com.qualcomm.robotcore.hardware.DcMotorSimple;


public class CDDriveChassis {
    CDHardware robotHardware;

    public CDDriveChassis(CDHardware theHardware) {
        robotHardware = theHardware;
        
        robotHardware.leftFrontMotor.setDirection(DcMotorSimple.Direction.FORWARD);
        robotHardware.leftRearMotor.setDirection(DcMotorSimple.Direction.FORWARD);
        robotHardware.rightFrontMotor.setDirection(DcMotorSimple.Direction.REVERSE);
        robotHardware.rightRearMotor.setDirection(DcMotorSimple.Direction.REVERSE);
        
        robotHardware.leftFrontMotor.setMode(DcMotor.RunMode.RUN_WITHOUT_ENCODER);
        robotHardware.leftRearMotor.setMode(DcMotor.RunMode.RUN_WITHOUT_ENCODER);
        robotHardware.rightFrontMotor.setMode(DcMotor.RunMode.RUN_WITHOUT_ENCODER);
        robotHardware.rightRearMotor.setMode(DcMotor.RunMode.RUN_WITHOUT_ENCODER);
    }

    public void setLeftFrontPower (double pow) {
        robotHardware.leftFrontMotor.setPower(pow);
    }
    public void setLeftRearPower (double pow) {
        robotHardware.leftRearMotor.setPower(pow);
    }
    public void setRightFrontPower (double pow) {
        robotHardware.rightFrontMotor.setPower(pow);
    }
    public void setRightRearPower (double pow) {
        robotHardware.rightRearMotor.setPower(pow);
    }

}

File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/CDDriveChassis.kt
package org.firstinspires.ftc.teamcode

// import com.qualcomm.robotcore.util.ElapsedTime
import com.qualcomm.robotcore.hardware.DcMotorSimple
import com.qualcomm.robotcore.hardware.DcMotor
// import com.qualcomm.robotcore.eventloop.opmode.Disabled;

class CDDriveChassis(var robotHardware: CDHardware) {
    // private val runtime = ElapsedTime()

    init {
        robotHardware.leftFrontMotor.direction = DcMotorSimple.Direction.FORWARD
        robotHardware.leftRearMotor.direction = DcMotorSimple.Direction.FORWARD
        robotHardware.rightFrontMotor.direction = DcMotorSimple.Direction.REVERSE
        robotHardware.rightRearMotor.direction = DcMotorSimple.Direction.REVERSE
        robotHardware.leftFrontMotor.mode = DcMotor.RunMode.RUN_WITHOUT_ENCODER
        robotHardware.leftRearMotor.mode = DcMotor.RunMode.RUN_WITHOUT_ENCODER
        robotHardware.rightFrontMotor.mode = DcMotor.RunMode.RUN_WITHOUT_ENCODER
        robotHardware.rightRearMotor.mode = DcMotor.RunMode.RUN_WITHOUT_ENCODER
    }

    fun setLeftFrontPower(pow: Double) {
        robotHardware.leftFrontMotor.power = pow
    }

    fun setLeftRearPower(pow: Double) {
        robotHardware.leftRearMotor.power = pow
    }

    fun setRightFrontPower(pow: Double) {
        robotHardware.rightFrontMotor.power = pow
    }

    fun setRightRearPower(pow: Double) {
        robotHardware.rightRearMotor.power = pow
    }
}